src/curve_intersections.rs

```diff
@@ -562,6 +579,17 @@ mod tests {
             &CubicBez::new((0.0, 0.0), (2.5, 0.5), (-1.5, 0.5), (1.0, 0.0)),
             9,
         );
+        // THESE TESTS FAIL, WORKING ON FIXES
+        // do_test(
+        //     &CubicBez::new((0.0, 0.0), (0.3, -1.0), (0.7, -1.0), (1.0, 0.0)),
+        //     &CubicBez::new((0.0, 0.0), (0.3, -1.0), (0.7, -1.0), (1.0, 0.0)),
+        //     2,
+        // );
+        // do_test(
+        //     &CubicBez::new((0.0, 0.0), (0.3, -1.0), (0.7, -1.0), (1.0, 0.0)),
+        //     &CubicBez::new((0.0, 0.0), (0.3, -0.5), (0.7, -0.5), (1.0, 0.0)),
+        //     2,
+        // );
```

Comment:
I've been thinking about this. Perhaps we don't want end-end intersections, but we really don't want to miss legitimate intersections, so we'll have to think about how we handle that. Maybe it's worth just dealing with end-point intersections so that we don't miss read intersections, or maybe there's a way to not have endpoint intersections without missing real intersections.